lib/galaxy/webapps/galaxy/api/workflows.py

```diff
@@ -496,9 +495,6 @@ def build_module(self, trans, payload={}):
         """
         inputs = payload.get('inputs', {})
         module = module_factory.from_dict(trans, payload)
-        module_state = {}
-        populate_state(trans, module.get_inputs(), inputs, module_state, check=False)
-        module.recover_state(module_state)
         return {
             'label'             : inputs.get('__label', ''),
             'annotation'        : inputs.get('__annotation', ''),
```

Comment:
Should this use `tool_state` instead of `inputs` to prevent it from being returned empty?

Reply:
Ah, good point, let me try this.

Reply:
Actually I see now that for new modules we do need to call `populate_state` here, not sure though why. Also I think this should stay `inputs.get(...)` as when you update the label or annotations this is being included in the `inputs`. 

It's just the cloning step that behaves different and directly copies label and annotation into the new node. So in practice copying labels / annotations works.